app/main/views/drafts.py

```diff
@@ -86,6 +86,7 @@ def edit_draft_service(draft_id):
     ).first_or_404()
 
     draft.update_from_json(update_json)
+    draft.purge_nulls()
```

Comment:
I think this could be part of `.data` validator. Both require us to disallow nulls as field values for service data, so I think we should make it an explicit rule ("null key are removed from data") rather than a delete workaround ("to delete set the value to null"). Making sure nulls are never written to the database seems better than removing them on first update too.